src/redux/books/BookSlice.js

```diff
@@ -9,10 +28,14 @@ const booksSlice = createSlice({
     addBook: (state, action) => {
       state.push(action.payload);
     },
-    removeBook: (state, action) => state.filter((book) => book.id !== action.payload),
+    removeBook: (state, action) => {
+      const bookIndex = state.filter((book) => book.id !== action.payload);
+      if (bookIndex !== -1) {
+        state.splice(bookIndex, 1);
+      }
+    },
```

Comment:
- [ ] I highly recommend using the `filter()` method while removing the book instead of using the `splice()` method which mutates the array. You read [this documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) for more info on how to use the suggested method.

```suggestion
    removeBook: (state, action) => {
      state = state.filter((book) => book.id  !== action.payload);
      }
```